### DISCUSSION TITLE
How to return the table contents in C# rpc function

### DISCUSSION THREAD

**Discussion by markcouvaras** [supabase/supabase#17020] [Questions]
Hi,

It appears that in the C# library one cannot return the table from a function. The Rpc function only works with void or record. But if I want to return a table it always returns a blank array. Is there a way we can return the table?

As an example I am calling it like this synchronously

  Dictionary<string, object> dict = new Dictionary<string, object>{ };
  dict.Add("lid", 11);
  var my_table = supabase.Rpc("get_league_table", dict).Result;

When I try to ready my_table .Content it is always just blank

**Comment by steve-chavez** [comment]
Make sure RLS is not filtering the table results

**Comment by brightertools** [comment]
I have successfully returned data from a function by creating a function  like this:

```
----------------------------------------------------------------------------------------------------------------------------------------------
-- get_auth_users
--
-- Gets the list of users
---------------------------------------------------------------------------------------------------------------------------------------------
CREATE OR REPLACE FUNCTION get_auth_users()
RETURNS TABLE (id uuid, email varchar, raw_app_meta_data jsonb)
SECURITY definer
AS $$
BEGIN
  RETURN QUERY SELECT au.id, au.email, au.raw_app_meta_data FROM auth.users au ;
END;
$$ LANGUAGE plpgsql;
```

C# code:

`var response = await _supabase.Rpc("get_auth_users", null);`
